api/src/main/scala/hmda/api/http/institutions/SubmissionPaths.scala

```diff
@@ -115,30 +115,69 @@ trait SubmissionPaths
       val path = s"institutions/$institutionId/filings/$period/submissions/$seqNr/edits"
       val submissionId = SubmissionId(institutionId, period, seqNr)
       extractExecutionContext { executor =>
-        implicit val ec: ExecutionContext = executor
-        val supervisor = system.actorSelection("/user/supervisor")
-        val fActor = (supervisor ? FindProcessingActor(HmdaFileValidator.name, submissionId)).mapTo[ActorRef]
-
-        val fEditChecks = for {
-          a <- fActor
-          f <- (a ? GetState).mapTo[HmdaFileValidationState]
-        } yield f
-
-        val fSummaryEdits = fEditChecks.map { editChecks =>
-          val s = validationErrorsToEditResults(editChecks.syntactical, Syntactical)
-          val v = validationErrorsToEditResults(editChecks.validity, Validity)
-          val q = validationErrorsToEditResults(editChecks.quality, Quality)
-          val m = validationErrorsToEditResults(editChecks.`macro`, Macro)
-          SummaryEditResults(s, v, q, m)
-        }
+        timedGet {
+          implicit val ec: ExecutionContext = executor
+          val supervisor = system.actorSelection("/user/supervisor")
+          val fActor = (supervisor ? FindProcessingActor(HmdaFileValidator.name, submissionId)).mapTo[ActorRef]
+
+          val fEditChecks = for {
+            a <- fActor
+            f <- (a ? GetState).mapTo[HmdaFileValidationState]
+          } yield f
+
+          val fSummaryEdits = fEditChecks.map { editChecks =>
+            val s = validationErrorsToEditResults(editChecks.syntactical, Syntactical)
+            val v = validationErrorsToEditResults(editChecks.validity, Validity)
+            val q = validationErrorsToEditResults(editChecks.quality, Quality)
+            val m = validationErrorsToEditResults(editChecks.`macro`, Macro)
+            SummaryEditResults(s, v, q, m)
+          }
 
-        onComplete(fSummaryEdits) {
-          case Success(edits) =>
-            complete(ToResponseMarshallable(edits))
-          case Failure(error) =>
-            completeWithInternalError(path, error)
+          onComplete(fSummaryEdits) {
+            case Success(edits) =>
+              complete(ToResponseMarshallable(edits))
+            case Failure(error) =>
+              completeWithInternalError(path, error)
+          }
         }
       }
 
     }
+
+  def submissionSingleEditPath(institutionId: String) =
+    path("filings" / Segment / "submissions" / IntNumber / "edits" / Segment) { (period, submissionId, editType) =>
```

Comment:
I think `seqNr` (as it's called in the edits endpoint) is a more accurate name for that variable than `submissionId`.